h/api/models.py

```diff
@@ -166,6 +175,113 @@ def title(self):
             return ""
 
     @property
+    def filename(self):
+        if is_file(self.uri):
+            return self.uri.split("/")[-1] or ""
+        else:
+            return ""
+
+    @property
+    def document_link(self):
```

Comment:
Do I need to setup the database model to test these?

But yes we could move it out of the model. If we did so it might be convenient if `search.search()` returned `AnnotationPresenter` objects, or maybe it's better if user code (such as view code) wraps `Annotation`s in `AnnotationPresenter`s itself.